### ISSUE TITLE
Library does not work with recent Servlet containers

### ISSUE THREAD

**Issue by uap-universe** [auth0/auth0-java-mvc-common#111]
### Describe the problem you'd like to have solved

The library does not work with recent Servlet containers that implement Jakarta EE.

### Describe the ideal solution

A new major version of this library should be made available, that depends on jakarta.servlet instead of javax.servlet.

Whoever wants to use Java EE Servlet containers, can rely on the current major version of this library.

## Alternatives and current work-arounds

Currently there does not seem to be an alternative in the com.auth0 group that could be used as a replacement for this library.

### Additional information, if any

Jakarta EE was released more than two years ago and is broadly adopted.

**Comment by jimmyjames** [comment]
👋 Thanks for raising @uap-universe! I agree that a new major should be made available, which would (among other things) target Jakarta EE (will need to decide which version to target, likely 9.1). We do have plans for a new major version; it will likely be 2023 as first we are going to make a new major of `auth0-java`. Thanks again!

**Comment by sethcrothers** [comment]
Also chiming in requesting support for the new version(s) of the Java servlet spec, specifically 5.0 which changes javax.servlet.* to jakarta.servlet.*.

**Comment by javad-md** [comment]
We also require support for the updated servlet spec - any update on this?

**Comment by teefal** [comment]
Also would like this as well.  Even just s/javax/jakarta :)

**Comment by siropa** [comment]
Any update for this? Need this in order to use with tomcat 10

**Comment by McCul** [comment]
Any update ?

**Comment by marvec** [comment]
Still nothing?

**Comment by rvinzent** [comment]
This issue is now well over a year old, are there any updates on a timeline here? It's starting to be a blocking dependency to upgrading other projects to LTS versions

**Comment by uap-universe** [comment]
It's over *four* years now, that Jakarta EE has been released, if I might add. The first versions of Tomcat that migrated to Jakarta EE are already end of life.

**Comment by rvinzent** [comment]
> We do have plans for a new major version; it will likely be 2023 as first we are going to make a new major of auth0-java

@jimmyjames any updates as we enter the second quarter of 2024?

**Comment by lakshyabatman** [comment]
I guess we have to move on? Folks have you figured out a way to handle this ?

I was thinking to make an adapter.

**Comment by cowwoc** [comment]
Honestly, don't bother. This library is dead. I ended up writing my own implementation for the REST API methods I needed, and it works just fine. auth0 should pull this library to stop wasting people's time.

**Comment by uap-universe** [comment]
> I guess we have to move on? Folks have you figured out a way to handle this ?

Yes, I removed it - and the entire Okta SSO solution - from my application. But I see this is not an easy step for everyone.

**Comment by tanya732** [comment]
Hi There,

Thank you for your patience. This has been noted and planned in this/coming quarter. Will keep you posted on further updates as we make progress.

Thank you

**Comment by jgritman** [comment]
I have inherited a codebase with a forked version of this repo to handle the jakarta changes. It is...painful. Good to hear you're making progress on the update so we could use the OSS version.

**Comment by VijayaNamboothiriTR** [comment]
Any updates? When do we get the new jar which supports recent servlet containers?

**Comment by tanya732** [comment]
Hi everyone 👋 , 

Thank you all for your patience, we know this has been a long-awaited update.

We're happy to share that the v2 major release (currently in beta) is now available, [here](https://github.com/auth0/auth0-java-mvc-common/releases/tag/2.0.0-beta.0). This release migrates the library to the Jakarta namespace (jakarta.servlet 6.0) and targets Java 17+, making it compatible with modern application servers and frameworks, including Tomcat 10.1+, Jetty 12+, WildFly 27+, and Spring Boot 3.x.

The migration guide and v2 branch are available [here](https://github.com/auth0/auth0-java-mvc-common/blob/v2/MIGRATION_GUIDE.md)

We appreciate everyone who raised this issue, provided feedback, and continued to follow along while we worked through the update. Please let us know if you encounter any issues while testing the beta release 🙏

**Comment by tanya732** [comment]
Hi everyone! 👋

Just a quick update on this issue: we've released mvc-auth-commons 2.0.0-beta.0, [here](https://github.com/auth0/auth0-java-mvc-common/releases/tag/2.0.0-beta.0), which includes Jakarta Servlet support.

If you get a chance to try it, we'd really appreciate hearing how it goes and if you run into any issues. If there are no further issues raised, we'll plan to close this issue in the coming week.

@jgritman, thanks again for maintaining your Jakarta-compatible fork while this work was in progress. If you happen to try the beta at some point, we'd be grateful for any feedback you have.

Migration guide: https://github.com/auth0/auth0-java-mvc-common/blob/2.0.0-beta.0/MIGRATION_GUIDE.md

Announcement & beta discussion: https://github.com/auth0/auth0-java-mvc-common/discussions/250

We're hoping to ship the stable release in about a month, so any feedback before then would be greatly appreciated. 

Thanks to everyone who's followed this issue and shared feedback along the way. It really helped us prioritize this work. 🙏

**Comment by tanya732** [comment]
Hi everyone! 👋

Following up with an update: **v2 is now available** (latest: **2.0.0-beta.2**) 🎉

This delivers the Jakarta support originally requested in this issue, closing it for now. The stable **2.0.0 GA** release is coming soon. If you've had a chance to try the beta, I'd love to hear your feedback!

Helpful links:

* **Release:** [2.0.0-beta.2](https://github.com/auth0/auth0-java-mvc-common/releases/tag/2.0.0-beta.2)
* **Beta discussion:** [here](https://github.com/auth0/auth0-java-mvc-common/discussions/250)

If you run into any issues or have additional feedback, please comment here and we'll be happy to reopen the discussion. Thanks again for your patience and support throughout this effort! 🙏
